common/pango.c

```diff
@@ -8,21 +8,72 @@
 #include <string.h>
 #include "log.h"
 
+static size_t escaped_text_length(const char *text) {
```

Comment:
Can you merge these two functions, so that if we change one value we don't forget to change the other?

(ie. if `src` is NULL, don't try to write to it, and always return the size)